projects/CuraEngine.cmake

```diff
@@ -17,6 +17,8 @@ elseif (BUILD_OS_OSX)
     endif()
 endif()
 
+list(APPEND extra_cmake_args -DCMAKE_CXX_FLAGS="-std=c++17")
```

Comment:
These custom C++ flags shouldn't be necessary as long as the `CMAKE_CXX_STANDARD` variable is set to 17. At least not since CMake 3.10. See also https://gitlab.kitware.com/cmake/cmake/-/issues/16482 .